main.tf

```diff
@@ -0,0 +1,94 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+resource "random_id" "id" {
+  byte_length = 16
+}
+
+resource "aws_dynamodb_table" "default" {
+  name           = "${var.application}-${var.environment-name}"
+  read_capacity  = "${var.autoscale_min_read_capacity}"
+  write_capacity = "${var.autoscale_min_write_capacity}"
+  hash_key       = "${var.hash_key}"
+  range_key      = "${var.range_key}"
+
+  attribute {
+    name = "${var.hash_key}"
+    type = "${var.hash_key_type}"
+  }
+
+  attribute {
+    name = "${var.range_key}"
+    type = "${var.range_key_type}"
+  }
+
+  server_side_encryption {
+    enabled = "${var.enable_encryption}"
+  }
+
+  lifecycle {
+    ignore_changes = ["read_capacity", "write_capacity"]
+  }
+
+  tags {
+    business-unit          = "${var.business-unit}"
+    application            = "${var.application}"
+    is-production          = "${var.is-production}"
+    environment-name       = "${var.environment-name}"
+    owner                  = "${var.team_name}"
+    infrastructure-support = "${var.infrastructure-support}"
+  }
+}
+
+module "dynamodb_autoscaler" {
```

Comment:
As discussed, I'm not sure I agree with calling this third-party module inside our module and perhaps this is something to be part of another example (first call our module then call this if you need autoscaling). I would like to see what the rest of @ministryofjustice/webops have to say?